environments/se/core/src/main/java/org/jboss/weld/environment/se/ExceptionHandler.java

```diff
@@ -0,0 +1,20 @@
+package org.jboss.weld.environment.se;
+
+import java.util.stream.Stream;
+
+public interface ExceptionHandler {
+
+    Stream<Class<? extends Throwable>> getExceptionTypes();
+
+    void handle(Throwable t);
```

Comment:
I would simplify the API so that `ExceptionHandler` only declares `boolean handle(Throwable t)` which returns true if the handler is able to handle the given exception.

Reply:
I would lean towards two methods, `supports()` and `handle()`, much like it's designed in [Junit](https://junit.org/junit5/docs/5.0.2/api/org/junit/jupiter/api/extension/ParameterResolver.html) for instance.

Reply:
The JUnit 5 ``ParameterResolver`` was actually my inspiration for this interface ... I've simplified it a bit now but the user will be responsible for all the ``supports()`` logic now.

Reply:
> Any chance that weld-core can be setup to also run JUnit 5 tests?

That shouldn't be problem, it can be run in parallel with JUnit 4, though I am not sure what is the actual benefit?
Plus any more complex testing that pure unit testing has to stick with JUnit 4 because Arquillian doesn't support it ATM (at least last time I checked it didn't).